File: client/nui/extras.ts
import { onNuiCallback } from "@trippler/tr_lib/client"
import { ToggleDamageCheck } from "../modules/extras"
import { getState } from "../states"

onNuiCallback<{ num: number }>('toggle', (data, cb) => {
  if (!getState.vehicleBusy) return
  setTimeout(() => {
    const extraNum = data.num || 1
    const isExtraOn = IsVehicleExtraTurnedOn(getState.vehicleHandle, extraNum)
    ToggleDamageCheck(isExtraOn, extraNum, cb)
  })
})

File: lenixm/lenix_patrol/client/nui/extras.lua
local extras<const> = require 'client/modules/extras'
local state<const> = require 'client/states'

lib.onNuiCallback('toggle', function(data, cb)
  if not state.getState.vehicleBusy then return end

  SetTimeout(0, function()
    local extraNum<const> = data.num or 1
    local isExtraOn<const> = IsVehicleExtraTurnedOn(state.getState.vehicleHandle, extraNum)
    extras.ToggleDamageCheck(isExtraOn, extraNum, cb)
  end)
end)